aws_topology/stackstate_checks/aws_topology/flowlogs.py

```diff
@@ -0,0 +1,340 @@
+                        # TODO there is the log record also contains "action" (ACCEPT or REJECT)
+                        if status != "NODATA" and status != "SKIPDATA" and nwitf:
+                            self.process_record(connections, log, nwitf)
+                else:
+                    self.log.warning("Flowlog had unsupported format")
+        return connections
+
+    def process_connection(self, id, connection):
+        network_interfaces = []
+        for _, itf in connection.network_interfaces.items():
+            network_interfaces.append(itf.original_data)
+        data = {
+            "traffic_type": connection.traffic_type,
+            "bytes_sent": connection.bytes_sent,
+            "bytes_received": connection.bytes_received,
+            "family": connection.family,
+            "bytes_sent_per_second": connection.bytes_sent_per_second,
+            "bytes_received_per_second": connection.bytes_received_per_second,
+            "local_address": connection.laddr,
+            "remote_address": connection.raddr,
+            "log": connection.traffic_log
+        }
+        data["network_interfaces"] = network_interfaces
+        data = make_valid_data(data)
+        # create component for local side
+        ip = connection.laddr.split(":")[0]
+        nwitf = connection.network_interfaces.get(ip, None)
+        lcid = "local/{}".format(id)
+        urns = [ipaddress_to_urn(ip, connection.namespace)]
+        if nwitf and nwitf.Association and nwitf.Association.PublicIp:
+            urns.append(ipaddress_to_urn(nwitf.Association.PublicIp, ""))
+        self.agent.component(
+            self.location_info,
+            lcid,
+            "vpc.request",
+            {"URN": urns, "debug_data": data},
+        )
+        # remote component for remote side
+        ip = connection.raddr.split(":")[0]
+        nwitf = connection.network_interfaces.get(ip, None)
+        rcid = "remote/{}".format(id)
+        urns = [ipaddress_to_urn(ip, connection.namespace)]
+        if nwitf and nwitf.Association and nwitf.Association.PublicIp:
+            urns.append(ipaddress_to_urn(nwitf.Association.PublicIp, ""))
+        self.agent.component(
+            self.location_info,
+            rcid,
+            "vpc.request",
+            {"URN": urns, "debug_data": data},
+        )
+        # make relation between the two
+        self.log.info(json.dumps(data, indent=2, default=str))
+        self.agent.relation(lcid, rcid, "flowlog", data)
+
+    def process_connections(self, connections):
+        count = 0
+        for id, connection in connections.items():
+            if should_process(connection):
+                self.process_connection(id, connection)
+                count += 1
+        return count
+
+    def _is_gz_file(self, body):
+        with io.BytesIO(body) as test_f:
+            return test_f.read(2) == b"\x1f\x8b"
+
+    def _get_stream(self, body):
```

Comment:
`_get_stream` is the same as the method in `cloudtrail.py`
`get_stream` and `is_gz_file` can stop being class methods and be functions in `utils.py` that are called from `flowlogs.py` and `cloudtrail.py` modules